### PR TITLE
Fix read example use io.CopyBuffer

### PR DESCRIPTION
**Pull request by nirs**
Due to zero copy optimizations in os.File, using io.CopyBuffer() with
os.File and qcow2reader.Image ends as io.Copy() ignoring our buffer.
This is slow and a bad example for users of the library.

Fix the issue by hiding os.File ReadFrom method with a wrapper, forcing
io.CopyBuffer() to use our buffer.

This speeds up the copy significantly:
- Reading qcow2 image is 3.7 times faster
- Copying qcow2 image is 1.7 times faster
- Reading qcow2 compressed image is 1.05 times faster

Based on the benchmarks, update the default buffer size in the read exxample
to 2 MiB.

buffer-size-qcow2

Fixes: #41

### CODE REVIEW

File: cmd/go-qcow2reader-example/read.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func cmdRead(args []string) error {
 		flag.PrintDefaults()
 	}
 	fs.BoolVar(&debug, "debug", false, "enable printing debug messages")
-	fs.IntVar(&bufferSize, "buffer-size", 65536, "buffer size")
+	fs.IntVar(&bufferSize, "buffer-size", 512*1024, "buffer size")
```
Comment (AkihiroSuda) [inline cmd/go-qcow2reader-example/read.go:32]: This is a separate topic and should be another commit

File: cmd/go-qcow2reader-example/read.go [hunk 2]
```diff
@@ -67,7 +67,19 @@ func cmdRead(args []string) error {
...
+	w := &Writer{os.Stdout}
+
+	_, err = io.CopyBuffer(w, sr, buf)
 
 	return err
 }
+
+// Writer forces os.File to write using our buffer by hiding it's ReadFrom
+// method. io.CopyBuffer is up to XX times faster with this wrrapper.
+type Writer struct {
```
Comment (AkihiroSuda) [inline cmd/go-qcow2reader-example/read.go:79]: `Writer` isn't a descriptive name.

What about:
```suggestion
type hideReadFrom struct {
```

File: cmd/go-qcow2reader-example/read.go [hunk 3]
```diff
@@ -67,7 +67,19 @@ func cmdRead(args []string) error {
...
-	_, err = io.CopyBuffer(os.Stdout, sr, buf)
+	w := &Writer{os.Stdout}
+
+	_, err = io.CopyBuffer(w, sr, buf)
 
 	return err
 }
+
+// Writer forces os.File to write using our buffer by hiding it's ReadFrom
+// method. io.CopyBuffer is up to XX times faster with this wrrapper.
```
Comment (AkihiroSuda) [inline cmd/go-qcow2reader-example/read.go:78]: Did you want to fulfill `XX`?

File: cmd/go-qcow2reader-example/read.go [hunk 4]
```diff
@@ -67,7 +67,19 @@ func cmdRead(args []string) error {
...
 	sr := io.NewSectionReader(img, offset, length)
-	_, err = io.CopyBuffer(os.Stdout, sr, buf)
+	w := &Writer{os.Stdout}
+
+	_, err = io.CopyBuffer(w, sr, buf)
 
 	return err
 }
+
+// Writer forces os.File to write using our buffer by hiding it's ReadFrom
```
Comment (AkihiroSuda) [inline cmd/go-qcow2reader-example/read.go:77]: it's → its ?

File: cmd/go-qcow2reader-example/read.go (hunk 3 above)
Comment (AkihiroSuda) [reply cmd/go-qcow2reader-example/read.go:78]: Maybe just add a link to this PR (https://github.com/lima-vm/go-qcow2reader/pull/42) for the benchmark result and the context

Comment (nirs) [reply cmd/go-qcow2reader-example/read.go:78]: I'll link the pr.

File: cmd/go-qcow2reader-example/read.go (hunk 4 above)
Comment (nirs) [reply cmd/go-qcow2reader-example/read.go:77]: Not sure, too easy to get this wrong :-)

Comment (nirs) [reply cmd/go-qcow2reader-example/read.go:77]: Right, should be "its"

File: cmd/go-qcow2reader-example/read.go (hunk 1 above)
Comment (nirs) [reply cmd/go-qcow2reader-example/read.go]: Separated to another commit.

File: cmd/go-qcow2reader-example/read.go (hunk 2 above)
Comment (nirs) [reply cmd/go-qcow2reader-example/read.go]: Renamed.

File: cmd/go-qcow2reader-example/read.go (hunk 3 above)
Comment (nirs) [reply cmd/go-qcow2reader-example/read.go]: I updated the docs for the new name.

File: cmd/go-qcow2reader-example/read.go (hunk 4 above)
Comment (nirs) [reply cmd/go-qcow2reader-example/read.go]: Removed in current version.

**Review by AkihiroSuda — APPROVED** [review]
Thanks
